Sources/SmithyReadWrite/DataProviding.swift

```diff
@@ -0,0 +1,13 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import struct Foundation.Data
+
+public protocol WireDataProviding: AnyObject {
```

Comment:
This protocol is provided so that HttpResponse can be used with SmithyReadWrite, without SmithyReadWrite having to depend on the HTTP implementation.